go-controller/hack/lint.sh

```diff
@@ -1,6 +1,10 @@
 #!/usr/bin/env bash
 
-GO111MODULE=on golangci-lint run \
+GOPATH=${GOPATH:-$(go env GOPATH)}
+if [[ ! -x "${GOPATH}/bin/golangci-lint" ]]; then
+  curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b ${GOPATH}/bin
```

Comment:
@dcbw how do we now go to the latest release of golanci-lint? don't we be stuck with the current version of v1.20.1?

Reply:
@girishmg good question; but I'm not sure how the original Makefile one got the latest either? Either that or it always downloaded it, which is what Casey said he didnt want.

So I really don't know what I'm supposed to do here. Either we always download the linter, or we only download it once.

(If Casey is talking about the 'go env GOPATH' well I disagree that running it every time is a problem, and he can just set GOPATH then)